src/ledger/events.rs

```diff
@@ -0,0 +1,95 @@
+use crate::eth::primitives::Hash;
+
+/// Represents token transfers (debits and credits) associated with a specific Ethereum account within a single transaction.
+///
+/// The `account_address` field identifies the main account involved in these transfers.
+///
+/// An event will be generated for each account involved in a transaction, meaning if a transaction is not the main account in this event,
+/// in another event it will be and credit and debit parties will be adjusted accordingly.
+pub struct AccountTransfersEvent {
+    /// ID of the event publication.
+    ///
+    /// If the event is republished, a new ID will be generated for the publication.
+    ///
+    /// Format: UUID v7 (xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx)
+    pub publication_id: Uuid,
+
+    /// Datetime of the event publication.
+    ///
+    /// Format: ISO 8601
+    pub publication_datetime: DateTime<Utc>,
+
+    /// Idempotency key of the event payload (TBD: `account_address` + `transaction_hash`).
+    ///
+    /// It is unique for each distinct event, but consistent across retries or republications of the same event payload.
+    ///
+    /// Format: String (TBD)
+    pub idempotency_key: String,
+
+    /// Address of the account that is part of all account transfers.
+    ///
+    /// Format: Prefixed account address - 20 bytes - 0x1234567890123456789012345678901234567890
+    pub account_address: Address,
+
+    /// Hash of the Ethereum transaction that originated account transfers.
+    ///
+    /// Format: Prefixed transaction hash - 32 bytes - 0x1234567890123456789012345678901234567890123456789012345678901234
+    pub transaction_hash: Hash,
+
+    /// Address of the contract that originated account transfers.
+    ///
+    /// Format: Prefixed account address - 20 bytes - 0x1234567890123456789012345678901234567890
+    pub contract_address: Address,
+
+    /// Identifier of the Ethereum function that originated account transfers.
+    ///
+    /// Format: Prefixed function signature - 4 bytes - 0x12345678
+    pub function_id: [u8; 4],
+
+    /// Number of the block that originated account transfers.
+    ///
+    /// Format: Integer
+    pub block_number: BlockNumber,
+
+    /// Datetime of the block that originated account transfers.
+    ///
+    /// Format: ISO 8601
+    pub block_datetime: DateTime<Utc>,
+
+    /// List of account transfers that the related account_address is part of.
+    pub account_transfers: Vec<AccountTransfer>,
+}
+
+/// Represents a token transfer between a debit party and a credit party.
+pub struct AccountTransfer {
+    /// Address of the token contract that executed the transfer between `debit_party_address` and `credit_party_address`.
+    ///
+    /// It may differ from the main `contract_address` because any contract can execute transfers in token contracts.
+    ///
+    /// Format: Prefixed account address - 20 bytes - 0x1234567890123456789012345678901234567890
+    pub token_address: Address,
```

Comment:
Token information (BRLC or JIM) can be derived from this field.

I need to validate if there is a change that non-token contracts can emit some form of transfer events, but I think that is not possible.

Reply:
I think it is "possible", someone, someday might create a contract that emits its own Transfer event with the same signature, I don't think there is anything that would bar someone from doing so if they wanted.

Either way, we could filter those logs out since we know the addresses of JIM and BRLC. That way even if someone, someday for some reason decided that it would be a good idea to emit an event in the same format of the ERC20 Transfer event it wouldn't matter because we would know to only publish events that concerned either contract.

Reply:
@carneiro-cw As decided together with the statements team, the implementation in Stratus will not use these mappings we have **for events**, that includes determining if the contract is JIM or BRLC.

This way they don't have to wait for a Stratus change and deployment in case these addresses changes. So they will have to handle on their side if something is BRLC, JIM or something else.

So if anything emits an event with the Transfer signature, it will be handled as a transfer.